src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -307,7 +307,9 @@ var ReactDOMFiberComponent = {
     }
     if (namespaceURI === HTML_NAMESPACE) {
       if (__DEV__) {
-        var isCustomComponentTag = isCustomComponent(type, props, namespaceURI);
+        var isCustomComponentTag = isCustomComponent(type, props, {
```

Comment:
This is fragile. Somebody might assume `domElement` is always a `domElement` in `isCustomComponent`, and break it.

Reply:
I hear that. Is this structurally fragile, or do we just need to change a name?

Reply:
Pushed a commit that does this slightly differently.